Can't load CFPlugin for ColdFusion 2023 on Windows

**Describe the bug**
When the ColdFusion 2023 service is ran on Windows, the `sun.java.command` system property doesn't exist. 

This causes an NullPointerException to be thrown whilst loading the CFPlugin as the isActive method will use Utils.isCFServer to check that the property contains `coldfusion`.   

This is a bigger problem since Deep's plugin loader changed as the previous plugin loader would have caught this exception preventing an interruption to Deep's startup.

**To Reproduce**
Steps to reproduce the behavior:
1. Install CF 2023 service on Windows
2. Attempt to start Deep with CFPlugin (was found via FusionReactor)

```
----------------------------------------------------------
ERROR: Failed to start deep agent: java.lang.reflect.InvocationTargetException null
----------------------------------------------------------
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.intergral.deep.agent.Agent.startNv(Agent.java:84)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.intergral.fusionreactor.deep.DeepComponent.loadDeep(DeepComponent.java:170)
        at com.intergral.fusionreactor.deep.DeepComponent.activate(DeepComponent.java:86)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.apache.felix.scr.impl.helper.BaseMethod.invokeMethod(BaseMethod.java:222)
        at org.apache.felix.scr.impl.helper.BaseMethod.access$500(BaseMethod.java:37)
        at org.apache.felix.scr.impl.helper.BaseMethod$Resolved.invoke(BaseMethod.java:615)
        at org.apache.felix.scr.impl.helper.BaseMethod.invoke(BaseMethod.java:499)
        at org.apache.felix.scr.impl.helper.ActivateMethod.invoke(ActivateMethod.java:295)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:302)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:113)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:847)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:814)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:763)
        at org.apache.felix.framework.ServiceRegistrationImpl.getFactoryUnchecked(ServiceRegistrationImpl.java:347)
        at org.apache.felix.framework.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:247)
        at org.apache.felix.framework.ServiceRegistry.getService(ServiceRegistry.java:350)
        at org.apache.felix.framework.Felix.getService(Felix.java:3737)
        at org.apache.felix.framework.BundleContextImpl.getService(BundleContextImpl.java:470)
        at org.apache.felix.scr.impl.manager.SingleRefPair.getServiceObject(SingleRefPair.java:72)
        at org.apache.felix.scr.impl.helper.BindMethod.getServiceObject(BindMethod.java:646)
        at org.apache.felix.scr.impl.manager.DependencyManager.getServiceObject(DependencyManager.java:2130)
        at org.apache.felix.scr.impl.manager.DependencyManager.doInvokeBindMethod(DependencyManager.java:1644)
        at org.apache.felix.scr.impl.manager.DependencyManager.invokeBindMethod(DependencyManager.java:1629)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeBindMethod(SingleComponentManager.java:370)
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:319)
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:295)
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1215)
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1136)
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:945)
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:881)
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1167)
        at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:120)
        at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990)
        at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838)
        at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545)
        at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4595)
        at org.apache.felix.framework.Felix.registerService(Felix.java:3587)
        at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:348)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:869)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:857)
        at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:133)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:915)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:715)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:399)
        at org.apache.felix.scr.impl.config.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:676)
        at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:339)
        at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:360)
        at org.apache.felix.scr.impl.Activator.access$000(Activator.java:53)
        at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:260)
        at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:259)
        at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:232)
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:479)
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:414)
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:232)
        at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:443)
        at org.apache.felix.framework.EventDispatcher.invokeBundleListenerCallback(EventDispatcher.java:915)
        at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:834)
        at org.apache.felix.framework.EventDispatcher.fireBundleEvent(EventDispatcher.java:516)
        at org.apache.felix.framework.Felix.fireBundleEvent(Felix.java:4579)
        at org.apache.felix.framework.Felix.startBundle(Felix.java:2174)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1373)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "String.contains(java.lang.CharSequence)" because the return value of "java.lang.System.getProperty(String)" is null
        at com.intergral.deep.plugin.cf.Utils.isCFServer(Utils.java:36)
        at com.intergral.deep.plugin.cf.CFPlugin.isActive(CFPlugin.java:54)
        at com.intergral.deep.agent.plugins.PluginSpiLoader.lambda$loadPlugins$1(PluginSpiLoader.java:62)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at com.intergral.deep.agent.plugins.PluginSpiLoader.loadPlugins(PluginSpiLoader.java:67)
        at com.intergral.deep.agent.DeepAgent.start(DeepAgent.java:75)
        at com.intergral.deep.agent.AgentImpl.startup(AgentImpl.java:56)
        ... 75 more

```

**Expected behavior**
Deep fails to start.

When started via the FusionReactor integration, FusionReactor will wait for Deep to be started, causing log spam and potentially blocking FusionReactor from starting fully.

```
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.intergral.deep.api.DeepAPILoader.loadAPI(DeepAPILoader.java:48)
        at com.intergral.deep.api.DeepAPILoader.awaitAPI(DeepAPILoader.java:74)
        at com.intergral.fusionreactor.deep.DeepComponent.loadDeep(DeepComponent.java:173)
        at com.intergral.fusionreactor.deep.DeepComponent.activate(DeepComponent.java:86)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.apache.felix.scr.impl.helper.BaseMethod.invokeMethod(BaseMethod.java:222)
        at org.apache.felix.scr.impl.helper.BaseMethod.access$500(BaseMethod.java:37)
        at org.apache.felix.scr.impl.helper.BaseMethod$Resolved.invoke(BaseMethod.java:615)
        at org.apache.felix.scr.impl.helper.BaseMethod.invoke(BaseMethod.java:499)
        at org.apache.felix.scr.impl.helper.ActivateMethod.invoke(ActivateMethod.java:295)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:302)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:113)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:847)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:814)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:763)
        at org.apache.felix.framework.ServiceRegistrationImpl.getFactoryUnchecked(ServiceRegistrationImpl.java:347)
        at org.apache.felix.framework.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:247)
        at org.apache.felix.framework.ServiceRegistry.getService(ServiceRegistry.java:350)
        at org.apache.felix.framework.Felix.getService(Felix.java:3737)
        at org.apache.felix.framework.BundleContextImpl.getService(BundleContextImpl.java:470)
        at org.apache.felix.scr.impl.manager.SingleRefPair.getServiceObject(SingleRefPair.java:72)
        at org.apache.felix.scr.impl.helper.BindMethod.getServiceObject(BindMethod.java:646)
        at org.apache.felix.scr.impl.manager.DependencyManager.getServiceObject(DependencyManager.java:2130)
        at org.apache.felix.scr.impl.manager.DependencyManager.doInvokeBindMethod(DependencyManager.java:1644)
        at org.apache.felix.scr.impl.manager.DependencyManager.invokeBindMethod(DependencyManager.java:1629)
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeBindMethod(SingleComponentManager.java:370)
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:319)
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:295)
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1215)
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1136)
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:945)
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:881)
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1167)
        at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:120)
        at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990)
        at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838)
        at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545)
        at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4595)
        at org.apache.felix.framework.Felix.registerService(Felix.java:3587)
        at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:348)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:869)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:857)
        at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:133)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:915)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:715)
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:399)
        at org.apache.felix.scr.impl.config.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:676)
        at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:339)
        at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:360)
        at org.apache.felix.scr.impl.Activator.access$000(Activator.java:53)
        at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:260)
        at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:259)
        at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:232)
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:479)
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:414)
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:232)
        at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:443)
        at org.apache.felix.framework.EventDispatcher.invokeBundleListenerCallback(EventDispatcher.java:915)
        at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:834)
        at org.apache.felix.framework.EventDispatcher.fireBundleEvent(EventDispatcher.java:516)
        at org.apache.felix.framework.Felix.fireBundleEvent(Felix.java:4579)
        at org.apache.felix.framework.Felix.startBundle(Felix.java:2174)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1373)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Must start DEEP first
        at com.intergral.deep.agent.AgentImpl.loadDeepAPI(AgentImpl.java:83)
        ... 72 more

```

**Screenshots**
![Screenshot from 2024-03-11 12-37-09](https://github.com/intergral/deep-java-client/assets/34251158/cd3cab23-245f-4897-9235-5fb2e5a56706)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+# 1.2.1 - (12/03/2024)
+- **[BUGFIX]**: plugin: fix ColdFusion detection on Windows [#116](https://github.com/intergral/deep/pull/116) [@LMarkie](https://github.com/LMarkie) 
+
 # 1.2.0 - (06/02/2024)
 - **[CHANGE]**: change log config to allow better control of logging [#103](https://github.com/intergral/deep/pull/103) [@Umaaz](https://github.com/Umaaz)
 - **[CHANGE]**: change tracepoint logger to be a plugin [#106](https://github.com/intergral/deep/pull/106) [@Umaaz](https://github.com/Umaaz)

--- a/plugins/cf-plugin/pom.xml
+++ b/plugins/cf-plugin/pom.xml
@@ -41,6 +41,15 @@
       <version>1.0-SNAPSHOT</version>
       <scope>compile</scope>
     </dependency>
+
+    <!-- https://mvnrepository.com/artifact/org.junit-pioneer/junit-pioneer -->
+    <dependency>
+      <groupId>org.junit-pioneer</groupId>
+      <artifactId>junit-pioneer</artifactId>
+      <version>2.2.0</version>
+      <scope>test</scope>
+    </dependency>
+
   </dependencies>
 
 </project>

--- a/plugins/cf-plugin/src/main/java/com/intergral/deep/plugin/cf/Utils.java
+++ b/plugins/cf-plugin/src/main/java/com/intergral/deep/plugin/cf/Utils.java
@@ -28,7 +28,7 @@ public final class Utils {
   /**
    * Are we running on a CF server.
    * <p>
-   * By checking that the coldfusion.home system property exists, or by looking at the java start up command,
+   * By checking that the {@code coldfusion.home} system property exists, or by looking at the java start up command,
    * we can tell if this is a CF server.
    *
    * @return {@code true} if we are on a coldfusion server.

--- a/plugins/cf-plugin/src/main/java/com/intergral/deep/plugin/cf/Utils.java
+++ b/plugins/cf-plugin/src/main/java/com/intergral/deep/plugin/cf/Utils.java
@@ -28,12 +28,22 @@ public final class Utils {
   /**
    * Are we running on a CF server.
    * <p>
-   * By looking at the java start up command we can tell if this is a CF server.
+   * By checking that the coldfusion.home system property exists, or by looking at the java start up command,
+   * we can tell if this is a CF server.
    *
    * @return {@code true} if we are on a coldfusion server.
    */
   public static boolean isCFServer() {
-    return System.getProperty("sun.java.command").contains("coldfusion");
+    if (System.getProperty("coldfusion.home") != null) {
+      return true;
+    }
+
+    final String javaCommand = System.getProperty("sun.java.command");
+    // has the potential to not exist on Windows services
+    if (javaCommand == null) {
+      return false;
+    }
+    return javaCommand.contains("coldfusion");
   }
 
   /**

--- a/plugins/cf-plugin/src/test/java/coldfusion/Version.java
+++ b/plugins/cf-plugin/src/test/java/coldfusion/Version.java
@@ -1,0 +1,27 @@
+/*
+ *     Copyright (C) 2024  Intergral GmbH
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Affero General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Affero General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Affero General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package coldfusion;
+
+public class Version {
+  public static int getMajor() {
+    if (System.getProperty("cf.test.error") != null) {
+      throw new RuntimeException("failed to load version");
+    }
+    return 10;
+  }
+}

--- a/plugins/cf-plugin/src/test/java/com/intergral/deep/plugin/cf/CFPluginTest.java
+++ b/plugins/cf-plugin/src/test/java/com/intergral/deep/plugin/cf/CFPluginTest.java
@@ -1,0 +1,82 @@
+/*
+ *     Copyright (C) 2024  Intergral GmbH
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Affero General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Affero General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Affero General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.intergral.deep.plugin.cf;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import com.intergral.deep.agent.api.plugin.EvaluationException;
+import com.intergral.deep.agent.api.plugin.ISnapshotContext;
+import com.intergral.deep.agent.api.resource.Resource;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class CFPluginTest {
+
+  @Test
+  void canDecorate() throws EvaluationException {
+    final CFPlugin cfPlugin = new CFPlugin();
+    final ISnapshotContext mock = Mockito.mock(ISnapshotContext.class);
+
+    final Resource decorate = cfPlugin.decorate(null, mock);
+    assertNotNull(decorate);
+    assertEquals("10", decorate.getAttributes().get("cf_version"));
+    assertNull(decorate.getAttributes().get("app_name"));
+
+    Mockito.verify(mock).evaluateExpression("APPLICATION.applicationname");
+  }
+
+  @Test
+  void canHandleEvaluateDecorate() throws EvaluationException {
+    final CFPlugin cfPlugin = new CFPlugin();
+
+    final ISnapshotContext mock = Mockito.mock(ISnapshotContext.class);
+
+    Mockito.when(mock.evaluateExpression("APPLICATION.applicationname")).thenReturn("app_name");
+
+    final Resource decorate = cfPlugin.decorate(null, mock);
+    assertNotNull(decorate);
+    assertEquals("10", decorate.getAttributes().get("cf_version"));
+    assertEquals("app_name", decorate.getAttributes().get("app_name"));
+
+    Mockito.verify(mock).evaluateExpression("APPLICATION.applicationname");
+  }
+
+  @Test
+  void canHandleEvaluateFailedDecorate() throws EvaluationException {
+    final CFPlugin cfPlugin = new CFPlugin();
+
+    final ISnapshotContext mock = Mockito.mock(ISnapshotContext.class);
+
+    Mockito.when(mock.evaluateExpression("APPLICATION.applicationname")).thenThrow(new RuntimeException("test exception"));
+
+    final Resource decorate = cfPlugin.decorate(null, mock);
+    assertNotNull(decorate);
+    assertEquals("10", decorate.getAttributes().get("cf_version"));
+    assertNull(decorate.getAttributes().get("app_name"));
+
+    Mockito.verify(mock).evaluateExpression("APPLICATION.applicationname");
+  }
+
+  @Test
+  void isActive() {
+    assertFalse(new CFPlugin().isActive());
+  }
+}

--- a/plugins/cf-plugin/src/test/java/com/intergral/deep/plugin/cf/UtilsTest.java
+++ b/plugins/cf-plugin/src/test/java/com/intergral/deep/plugin/cf/UtilsTest.java
@@ -32,6 +32,7 @@ class UtilsTest {
   void canDetectNotCF() {
     assertFalse(Utils.isCFServer());
   }
+
   @Test
   @ClearSystemProperty(key = "sun.java.command")
   void canHandleNoSunCommand() {
@@ -39,13 +40,13 @@ class UtilsTest {
   }
 
   @Test
-  @SetSystemProperty(key = "coldfusion.home", value ="doesn't matter")
+  @SetSystemProperty(key = "coldfusion.home", value = "doesn't matter")
   void canDetectCFHome() {
     assertTrue(Utils.isCFServer());
   }
 
   @Test
-  @SetSystemProperty(key = "sun.java.command", value ="/some/coldfusion")
+  @SetSystemProperty(key = "sun.java.command", value = "/some/coldfusion")
   void canUseSunCommand() {
     assertTrue(Utils.isCFServer());
   }
@@ -56,7 +57,7 @@ class UtilsTest {
   }
 
   @Test
-  @SetSystemProperty(key="cf.test.error", value = "anything")
+  @SetSystemProperty(key = "cf.test.error", value = "anything")
   void failLoadCFVersion() {
     assertNull(Utils.loadCFVersion());
   }

--- a/plugins/cf-plugin/src/test/java/com/intergral/deep/plugin/cf/UtilsTest.java
+++ b/plugins/cf-plugin/src/test/java/com/intergral/deep/plugin/cf/UtilsTest.java
@@ -1,0 +1,63 @@
+/*
+ *     Copyright (C) 2024  Intergral GmbH
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Affero General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Affero General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Affero General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.intergral.deep.plugin.cf;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import org.junitpioneer.jupiter.ClearSystemProperty;
+import org.junitpioneer.jupiter.SetSystemProperty;
+
+class UtilsTest {
+
+  @Test
+  void canDetectNotCF() {
+    assertFalse(Utils.isCFServer());
+  }
+  @Test
+  @ClearSystemProperty(key = "sun.java.command")
+  void canHandleNoSunCommand() {
+    assertFalse(Utils.isCFServer());
+  }
+
+  @Test
+  @SetSystemProperty(key = "coldfusion.home", value ="doesn't matter")
+  void canDetectCFHome() {
+    assertTrue(Utils.isCFServer());
+  }
+
+  @Test
+  @SetSystemProperty(key = "sun.java.command", value ="/some/coldfusion")
+  void canUseSunCommand() {
+    assertTrue(Utils.isCFServer());
+  }
+
+  @Test
+  void loadCFVersion() {
+    assertEquals("10", Utils.loadCFVersion());
+  }
+
+  @Test
+  @SetSystemProperty(key="cf.test.error", value = "anything")
+  void failLoadCFVersion() {
+    assertNull(Utils.loadCFVersion());
+  }
+}

--- a/pom.xml
+++ b/pom.xml
@@ -553,7 +553,8 @@
             <dependency>
                 <groupId>org.junit.jupiter</groupId>
                 <artifactId>junit-jupiter</artifactId>
-                <version>5.10.1</version>
+                <!-- Version upgrade blocked by junit-pioneer version -->
+                <version>5.9.3</version>
             </dependency>
             <dependency>
                 <groupId>org.mockito</groupId>

